GW2EIEvtcParser/EncounterLogic/Strikes/SotO/TempleOfFebe.cs

```diff
@@ -500,33 +500,119 @@ internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog l
                 }
             }
         }
-        
+
+        /// <summary>
+        /// Hide the Permanent Embodiments in Challenge Mode.<br></br>
+        /// The Embodiments are shown during their cast events.
+        /// </summary>
+        /// <param name="target">The casting Embodiment.</param>
+        /// <param name="log">The log.</param>
+        /// <param name="replay">The Combat Replay.</param>
         private static void AddHiddenWhileNotCasting(NPC target, ParsedEvtcLog log, CombatReplay replay)
         {
             var castEvents = target.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.SkillId != WeaponStow && x.SkillId != WeaponSwap && x.SkillId != WeaponDraw).ToList();
             long invisibleStart = log.FightData.LogStart;
             bool startTrimmed = false;
-            foreach (AbstractCastEvent e in castEvents)
+
+            // Cast durations of each mechanic with +- 20ms leeway
+            foreach (AbstractCastEvent cast in castEvents)
             {
+                int castDuration = 0;
+                switch (cast.SkillId)
+                {
+                    case CrushingRegretNM:
```

Comment:
Target species ID check would look easier on the eyes no? Would also make it possible to do it outside of the forEach.
Or is it possible for Embodiments to cast different skills? Like Despair casting Crushing Regret?

Reply:
It's not possible but also i've included normal mode skills which makes no sense, this is only for cm :D
Since we already have a target switch before calling the method, i could pass the cast duration in the parameters and discard the switch completely

Reply:
Should be better now